src/briefcase/integrations/java.py

```diff
@@ -310,8 +310,17 @@ def install(self):
 
     def uninstall(self):
         """Uninstall a JDK."""
-        with self.tools.console.wait_bar("Removing old JDK install..."):
-            if self.tools.host_os == "Darwin":
-                self.tools.shutil.rmtree(self.java_home.parent.parent)
-            else:
-                self.tools.shutil.rmtree(self.java_home)
+        try:
+            with self.tools.console.wait_bar("Removing old JDK install..."):
+                if self.tools.host_os == "Darwin":
+                    self.tools.shutil.rmtree(self.java_home.parent.parent)
+                else:
+                    self.tools.shutil.rmtree(self.java_home)
+        except PermissionError as e:
+            raise BriefcaseCommandError(
+                """\
+Permission denied when trying to remove Java.
```

Comment:
So - this is *a* fix for the problem I reported; but not the one I intended. That's my fault for not being explicit. 

It *is* desirable to give a specific location for the JDK that can't be removed - that way the user has the relevant detail to diagnose any possible problems. However, that location (and thus the message) is platform dependent. This means the location that you put in this error message needs to be conditional on the `self.tools.host_os` check a couple of lines above. Instead of just `rmtree(<some location>)` retain that location as a variable, `rmtree` that variable, and then include the variable in the error message.